Octopus Deploy's Log4J Response (#629)
OctopusDeploy/blog: blog/2021-12/octopus-deploy-log4j-response/index.md

---
title: Octopus Deploy's response to Log4J
description: Learn how Octopus Deploy responded to the Log4J vulnerability (CVE-2021-44228).
author: kyle.jackson@octopus.com
visibility: public
published: 2021-12-14-1400
metaImage: blogimage-security.png
bannerImage: blogimage-security.png
bannerImageAlt: Octopus on shield to represent security
isFeatured: false
tags: 
  - Product
  - Trust and Security
---

On Thursday, December 9, 2021, Octopus Deploy was made aware of a vulnerability in the Log4J logging utility ([CVE-2021-44228](https://cve.mitre.org/cgi-bin/cvename.cgi?name=2021-44228)). We immediately enacted our Security Incident Response Plan to investigate our use of this utility and its impact on our products and infrastructure.

Fortunately, Octopus Cloud, Octopus Server, and Octopus Tentacle are not affected. However, there are some related products that are impacted and require updates.

In this post, we share the results of our investigation and our recommended next steps for customers.

> ## JetBrains TeamCity plugin

Reviewer: Suggestion for specificity.

```suggestion
## JetBrains TeamCity plugin for Octopus Deploy
```

> Version 6.1.7 of the JetBrains TeamCity plugin has been updated to use version 2.15.0 of Log4J. However, all versions of the plugin before 6.1.7 are using version 2.14.1 of Log4J and could be vulnerable to remote code execution.

Reviewer: I didn't pick this up when editing the email copy 🤦 , but we updated the plugin _to_ v6.1.7. If we updated v6.1.7 then we'd have the next version number. This is a techy nitpicking detail that some people in our audience may notice.

```suggestion
Version 6.1.7 of the JetBrains TeamCity plugin uses version 2.15.0 of Log4J. However, all versions of the plugin before 6.1.7 are using version 2.14.1 of Log4J and could be vulnerable to remote code execution.
```

### What you need to do

> Version 6.1.7 of the Octopus Deploy plugin for TeamCity has been published to the marketplace. If you're a TeamCity user, you should be notified of this update through the Web UI and you're strongly encouraged to update.

Reviewer: Another option to omit the passive voice.

```suggestion
Version 6.1.7 of the Octopus Deploy plugin for TeamCity is available from the marketplace. If you're a TeamCity user, you should be notified of this update through the Web UI and you're strongly encouraged to update.
```

> Version 6.1.7 of the Octopus Deploy plugin for TeamCity has been published to the marketplace. If you're a TeamCity user, you should be notified of this update through the Web UI and you're strongly encouraged to update.

Reviewer: I've been trying to improve the second sentence, but I'm struggling. 

It's unclear whether the update notification is through the Octopus UI or a TeamCity UI. My brain also wants to see something at the end of the sentence. Perhaps `the plugin`.

Reviewer: Alternatively, you could end the sentence with something like `you're strongly encouraged to accept the update`.

Reviewer: Even better, less passive, `we strongly encourage you to accept the update`.

Author: Can we go with `we strongly encourage you to install the update` ?

Reviewer 2: Thanks @kathrynmarks and @wizedkyle - I've updated to `we strongly encourage you to install the update`

> Version 6.1.7 of the Octopus Deploy plugin for TeamCity is available from the marketplace. If you're a TeamCity user, you should be notified of this update through the Web UI and we strongly encourage you to accept the update.

Reviewer 2: ```suggestion
Version 6.1.7 of the Octopus Deploy plugin for TeamCity is available from the marketplace. If you're a TeamCity user, you should be notified of this update through the Web UI and we strongly encourage you to install the update.
```

> Version 6.1.7 of the Octopus Deploy plugin for TeamCity is available from the marketplace. If you're a TeamCity user, you should be notified of this update through the Web UI and you're strongly encouraged to update the plugin.

Reviewer 2: ```suggestion
Version 6.1.7 of the Octopus Deploy plugin for TeamCity is available from the marketplace. If you're a TeamCity user, you should be notified of this update through the Web UI and we strongly encourage you to accept the update.
```

Follow our [security advisory](https://advisories.octopus.com/adv/2021-12---Octopus-Deploy-TeamCity-Plugin-log4j2-dependency.2306410241.html) for updates.

## Octopus Java SDK

> Version 0.0.3 of the Java SDK has been updated to use version 2.15.0 of Log4J. However, all versions of the Java SDK before 0.0.3 are using version 2.14.1 of Log4J and could be vulnerable to remote code execution.

Reviewer: (as above)

```suggestion
Version 0.0.3 of the Java SDK uses version 2.15.0 of Log4J. However, all versions of the Java SDK before 0.0.3 are using version 2.14.1 of Log4J and could be vulnerable to remote code execution.
```
